src/utils/thread_helper.cpp

```diff
@@ -613,15 +630,17 @@ void ThreadHelper::AttachAllNodesTo(const std::vector<uint8_t> &aDatasetTlvs, At
         {
             mAttachDelayMs            = 0;
             mAttachPendingDatasetTlvs = {};
+            mWaitingMgmtSetResponse   = false;
```

Comment:
It seems this should be set to false in both `hasActiveDataset == true` and `hasActiveDataset == false` cases.